### ISSUE TITLE
omegaconf.errors.ValidationError: Unexpected object type: dict

### ISSUE THREAD

**Issue by SirJohnFranklin** [manujosephv/pytorch_tabular#85]
**Describe the bug**
Running the example code stops at line:
> tabular_model = TabularModel(...) 

because omegaconf does not expect dictionaries as keys, but by standard, optimizer_params is a dict of the OptimizerConfig.

**To Reproduce**
Create new environment and install all necessary tools including 
> pip install git+https://github.com/manujosephv/pytorch_tabular

at the end.

**Expected behavior**
TabularModel is created.

**Screenshots**
CODE:

```
tabular_model = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config,
    experiment_config=experiment_config,
)
Traceback (most recent call last):
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\IPython\core\interactiveshell.py", line 3398, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-e33239794d63>", line 49, in <cell line: 49>
    tabular_model = TabularModel(
  File "C:\Users\XXX\AppData\Roaming\Python\Python39\site-packages\pytorch_tabular\tabular_model.py", line 85, in __init__
    optimizer_config = self._read_parse_config(
  File "C:\Users\XXX\AppData\Roaming\Python\Python39\site-packages\pytorch_tabular\tabular_model.py", line 178, in _read_parse_config
    config = OmegaConf.structured(config)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 123, in structured
    return OmegaConf.create(obj, parent, flags)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 176, in create
    return OmegaConf._create_impl(
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 882, in _create_impl
    format_and_raise(node=None, key=None, value=None, msg=str(e), cause=e)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 793, in format_and_raise
    _raise(ex, cause)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 771, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 843, in _create_impl
    return DictConfig(
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 793, in format_and_raise
    _raise(ex, cause)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 771, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\dictconfig.py", line 94, in __init__
    self._set_value(content, flags=flags)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\dictconfig.py", line 649, in _set_value
    raise e
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\dictconfig.py", line 646, in _set_value
    self._set_value_impl(value, flags)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\dictconfig.py", line 676, in _set_value_impl
    data = get_structured_config_data(value, allow_objects=ao)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 481, in get_structured_config_data
    return get_dataclass_data(obj, allow_objects=allow_objects)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 409, in get_dataclass_data
    format_and_raise(
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 873, in format_and_raise
    _raise(ex, cause)
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 771, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\_utils.py", line 401, in get_dataclass_data
    d[name] = _maybe_wrap(
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 1070, in _maybe_wrap
    return _node_wrap(
  File "C:\Users\XXX\anaconda3\envs\XXX\lib\site-packages\omegaconf\omegaconf.py", line 1052, in _node_wrap
    raise ValidationError(f"Unexpected object type: {type_str(ref_type)}")
omegaconf.errors.ValidationError: Unexpected object type: dict
    full_key: optimizer_params
    object_type=OptimizerConfig
```

**Desktop (please complete the following information):**
 - OS: Windows
 - Python 3.9 or 3.10

**Comment by TuozhenLiu** [comment]
Meet this bug too

**Comment by TuozhenLiu** [comment]
> Meet this bug too

Follow this [reply](https://github.com/manujosephv/pytorch_tabular/issues/77#issuecomment-1132416246)! I fix the bug by "pip install omegaconf==2.1.2".

**Comment by manujosephv** [comment]
> > Meet this bug too
> 
> Follow this [reply](https://github.com/manujosephv/pytorch_tabular/issues/77#issuecomment-1132416246)! I fix the bug by "pip install omegaconf==2.1.2".

This is because of a breaking change in `omegaconf`.  Current fix is what @TuozhenLiu suggested. This wil be fixed the next release (it's already fixed in main branch)

**Comment by manujosephv** [comment]
closing this now
